docs/tutorials/model-registry.txt

```diff
@@ -0,0 +1,245 @@
+load the checkpoints they refer to. The following is an example JSON
+representation of a model version for illustration.
+
+.. code:: json
+
+  {
+    "model_version": {
+      "model": {
+        "name": "mnist_cnn",
+        "description": "a character recognition model",
+        "metadata": {
+          "dataset_url": "http://yann.lecun.com/exdb/mnist/",
+          "git_url": "http://github.com/user/repo"
+        },
+      },
+      "checkpoint": {
+        "uuid": "6a24d772-f1f7-4655-9061-22d582afd96c",
+        "experiment_config": { "...": "..." },
+        "experimentId": 1,
+        "trialId": 1,
+        "hparams": { "...": "..." },
+        "batchNumber": 100,
+        "resources": {
+          "code/": "0",
+          "code/const.yaml": "391",
+          "code/model_def.py": "4067",
+          "determined-keras-model.h5": "25224"
+        },
+        "metadata": {},
+        "framework": "tensorflow-1.14.0",
+        "format": "h5",
+        "metrics": { "...": "..." }
+      },
+      "version": 1,
+    }
+  }
+
+Methods accessing model versions often return a
+:class:`~determined.experimental.Checkpoint` since a model version is simply a
+pointer to a checkpoint. Therefore, the model_version can be used to download
+and load checkpoints as well as manage checkpoint metadata. See the
+:ref:`use-trained-models` guide for more information.
+
+The following snippet registers a version of a model and returns the updated
+:class:`~determined.experimental.Checkpoint` object.
+
+.. code:: python
+
+  from determined.experimental import Determined
+
+  d = Determined()
+
+  checkpoint = d.get_experiment(exp_id).top_checkpoint()
+
+  model = d.get_model("model_name")
+
+  model_version = model.register_version(checkpoint.uuid)
+
+  # Depending on the framework used to generate the checkpoint, the
+  # model_version will either load a TensorFlow or PyTorch object with the
+  # architecture and weights loaded from the checkpoint.
+  tf_or_torch_model = model_version.load()
+
+Similarly, to register a new model version from the CLI use the following
+command.
+
+.. code:: bash
+
+  det model register <model_name> <checkpoint-uuid>
+
+The following snippet retrieves a specific version of a model. If no version
+number is specified the most recent version of the model is returned.
+
+.. code:: python
+
+  from determined.experimental import Determined
+
+  model = Determined().get_model("model_name")
+
+  latest_version = model.get_version()
+
+  specific_versions = model.get_version(3)
+
+To list all the versions of a model use the following snippet. By default model
+versions are returned in descending order such that the most recent models are
+returned first.
+
+.. code:: python
+
+  from determined.experimental import Determined
+
+  model = Determined().get_model("model_name")
+
+  model_versions = model.get_versions()
+
+The CLI equivalent is as follows.
+
+.. code:: bash
+
+  det model list_versions <model_name>
```

Comment:
This isn't a documentation issue, but elsewhere we use hyphens to separate words in CLI commands, not underscores. (e.g., `det experiment list-trials`).

Reply:
I'm changing the documentation now and will open a PR to change the underscore to a hyphen.